src/core/worker.js

```diff
@@ -82,6 +84,342 @@ var WorkerTask = (function WorkerTaskClosure() {
   return WorkerTask;
 })();
 
+//#if !PRODUCTION
+/**
+ * Interface that represents PDF data transport. If possible, it allows
+ * progressively load entire or fragment of the PDF binary data.
+ *
+ * @interface
+ * */
+function IPDFStream() {}
+IPDFStream.prototype = {
+  /**
+   * Gets a reader for the entire PDF data.
+   * @returns {IPDFStreamReader}
+   */
+  getFullReader: function () { return null; },
+
+  /**
+   * Gets a reader for the range of the PDF data.
+   * @param {number} begin - the start offset of the data.
+   * @param {number} end - the end offset of the data.
+   * @returns {IPDFStreamRangeReader}
+   */
+  getRangeReader: function (begin, end) { return null; },
+
+  /**
+   * Cancels all opened reader and closes all their opened requests.
+   * @param {Object} reason - the reason for cancelling
+   */
+  cancelAllRequests: function (reason) {},
+};
+
+/**
+ * Interface for a PDF binary data reader.
+ *
+ * @interface
+ */
+function IPDFStreamReader() {}
+IPDFStreamReader.prototype = {
+  /** Gets promise when the headers and other meta information is known about
+   * the PDF binary data.
+   * @returns {Promise}
+   */
+  get headersReady() { return null; },
+
+  /**
+   * Gets PDF binary data length. It is defined after the headersReady promise
+   * is resolved.
+   * @returns {number} The data length (or 0 if unknown).
+   */
+  get contentLength() { return 0; },
+
+  /**
+   * Gets ability of the stream to handle range requests. It is defined after
+   * the headersReady promise is resolved.
+   * @returns {boolean}
+   */
+  get disableRange() { return false; },
```

Comment:
This is a weird name. What about `isRangeSupported`, or at least `rangeDisabled`? `disableRange` looks like something that can be set to disable range requests.

"Returns whether range requests are supported for this stream."

(or "partial range requests".)